TauAnalysis/MCEmbeddingTools/plugins/DYToMuTauGenFilter.cc

```diff
@@ -0,0 +1,133 @@
+}
+
+DYToMuTauGenFilter::~DYToMuTauGenFilter() {}
+
+bool DYToMuTauGenFilter::leptondecay(const reco::Candidate *d, int depth) {
+  // returns true if the event has an leptonic decay
+  //  Debug Output
+  //  std::cout << std::string(4*depth, '-') << "---Hadroncheck-" << std::endl;
+  //  std::cout << std::string(4*depth, '-') << "|  Depth       " << depth << std::endl;
+  //  std::cout << std::string(4*depth, '-') << "|  ID:         " << d->pdgId() << std::endl;
+  //  std::cout << std::string(4*depth, '-') << "|  Status:     " << d->status() << std::endl;
+  //  std::cout << std::string(4*depth, '-') << "|  NDaughters: " << d->numberOfDaughters()<< std::endl;
+  bool check = false;
+  if (d->status() != 1) {
+    if (d->numberOfDaughters() == 3) {
+      if (std::abs(d->daughter(0)->pdgId()) == 14 || std::abs(d->daughter(1)->pdgId()) == 14 ||
+          std::abs(d->daughter(2)->pdgId()) == 14 || std::abs(d->daughter(0)->pdgId()) == 12 ||
+          std::abs(d->daughter(1)->pdgId()) == 12 || std::abs(d->daughter(2)->pdgId()) == 12)
+        check = true;
+    } else if (d->numberOfDaughters() > 3)
+      return false;
+    if (d->numberOfDaughters() < 4) {
+      for (unsigned int k = 0; k < d->numberOfDaughters(); k++) {
+        // std::cout << std::string(4*depth, '-') << "| Daughter Number " << k << std::endl;
+        int new_depth = depth + 1;
+        if (leptondecay(d->daughter(k), new_depth) == true)
+          check = true;
+      }
+    }
+  }
+  return check;
+}
+bool DYToMuTauGenFilter::muondecay(const reco::Candidate *d, int depth) {
+  // returns true if the event has an muon decay
+  bool check = false;
+  if (d->status() != 1) {
+    if (d->numberOfDaughters() == 3) {
+      if (std::abs(d->daughter(0)->pdgId()) == 14 || std::abs(d->daughter(1)->pdgId()) == 14 ||
+          std::abs(d->daughter(2)->pdgId()) == 14)
+        check = true;
+    } else if (d->numberOfDaughters() > 3)
+      return false;
+    if (d->numberOfDaughters() < 4) {
+      for (unsigned int k = 0; k < d->numberOfDaughters(); k++) {
+        int new_depth = depth + 1;
+        if (muondecay(d->daughter(k), new_depth) == true)
+          check = true;
+      }
+    }
+  }
+  return check;
+}
+
+bool DYToMuTauGenFilter::filter(edm::Event &iEvent, const edm::EventSetup &iSetup) {
+  iEvent.getByToken(genParticleCollection_, gen_handle);
+
+  for (unsigned int i = 0; i < gen_handle->size(); i++) {
+    const reco::GenParticle gen_particle = (*gen_handle)[i];
+    // Check if Z Boson decayed into two leptons
+    if (gen_particle.pdgId() == 23 && gen_particle.numberOfDaughters() == 2) {
+      // Check if daugther particles are taus
+      // From Generator: Mu.Pt > 18 && Had.Pt > 25 && Mu.Eta < 2.1
+      if (std::abs(gen_particle.daughter(0)->pdgId()) == 15 && std::abs(gen_particle.daughter(0)->eta()) < 2.1 &&
+          gen_particle.daughter(0)->pt() > 25 && gen_particle.daughter(1)->pt() > 18) {
```

Comment:
more hard-coded cuts to possibly make configurable.

Reply:
File also removed in a8eae9f as it is not used anymore.